src/main/kotlin/com/levi/featurespitter/controller/StoreController.kt

```diff
@@ -0,0 +1,89 @@
+package com.levi.featurespitter.controller
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RestController
+
+import com.levi.featurespitter.model.Store
+import com.levi.featurespitter.model.StoreSeason
+import com.levi.featurespitter.repository.StoreRepository
+import com.levi.featurespitter.service.StoreService
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.RequestMapping
+import com.levi.featurespitter.service.EndPointService
+import com.levi.featurespitter.main
+import java.util.Collections
+import org.springframework.web.bind.annotation.CrossOrigin
+
+@RestController
+class StoreController(private val storeService: StoreService) {
+
+    @CrossOrigin
+	@GetMapping("/stores")
+	fun getAllStores(): List<Store> {
+
+		var allStores: List<Store> = storeService.findAll()
+
+		return allStores;
+	}
+
+    @CrossOrigin
+	@PutMapping("/store/{storeid}")
+	fun updateStoreName(@RequestBody store: Store, @PathVariable storeid: Long) : Store{
+		
+		println(store.toString())
+		return storeService.updateStore(store)
+	}
+
+	@GetMapping("/store/csv")
+	fun exportCSV() {
+		//TODO
```

Comment:
Was this later implemented in the Frontend? What would be the purpose of this endpoint?

Reply:
I wanted to implement it here and return the csv to frontend. But later, Instead of getting data from database and export, I directly created csv at the frontend after getting all the data from other backend api in app view.

Its purpose was to read whole data from stores table and return it in csv format.

Reply:
Having this in the frontend only works because you don't have pagination implemented in the backend, which can also create performance issues in the frontend, specially when dealing with millions of records.

Although, this could be easily achivable through the `Pageable` class in the spring repositories.

Reply:
Thanks for suggestion. Will take care of this, If got chance to work upon similar scenarios.